test/integration/test_sriov_cni.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+this_folder="$(dirname "$(readlink --canonicalize "${BASH_SOURCE[0]}")")"
+export CNI_PATH="${this_folder}/test_utils"
+export CNI_CONTAINERID=stub_container
+
+setup() {
+  ip netns del test_root_ns || true
+  ip netns add test_root_ns
+
+  # See pkg/utils/testing.go
+  ip netns exec test_root_ns ip link add enp175s0f1 type dummy
+  ip netns exec test_root_ns ip link add enp175s6 type dummy
+  ip netns exec test_root_ns ip link add enp175s7 type dummy
+
+  DEFAULT_CNI_DIR=$(mktemp -d)
+  export DEFAULT_CNI_DIR
+}
+
+teardown() {
+  # Double check the variable points to something created by the setup() function.
+  if [[ $DEFAULT_CNI_DIR == "/tmp/tmp."* ]]; then
+    rm -rf "$DEFAULT_CNI_DIR"
+  fi
+}
+
+test_macaddress() {
+
+  make_container "container_1"
+
+  export CNI_IFNAME=net1
+
+  read -r -d '' CNI_INPUT <<- EOM
+  {
+    "type": "sriov",
+    "cniVersion": "0.3.1",
+    "name": "sriov-network",
+    "ipam": {
+      "type": "test-ipam-cni"
+    },
+    "deviceID": "0000:af:06.0",
+    "mac": "60:00:00:00:00:E1",
+    "logFile": "${DEFAULT_CNI_DIR}/sriov.log",
+    "logLevel": "debug" 
+  }
+EOM
+
+  export CNI_COMMAND=ADD
+  assert invoke_sriov_cni
+  assert 'ip netns exec container_1_netns ip link | grep -i 60:00:00:00:00:E1'
+
+  export CNI_COMMAND=DEL
+  assert 'invoke_sriov_cni'
+  assert 'ip netns exec test_root_ns ip link show enp175s6'
+}
+
+
+test_vlan() {
+
+  make_container "container_1"
+
+  export CNI_IFNAME=net1
+
+  read -r -d '' CNI_INPUT <<- EOM
+  {
+    "type": "sriov",
+    "cniVersion": "0.3.1",
+    "name": "sriov-network",
+    "vlan": 1234,
+    "ipam": {
+      "type": "test-ipam-cni"
+    },
+    "deviceID": "0000:af:06.0",
+    "mac": "60:00:00:00:00:E1",
+    "logLevel": "debug"
+  }
+EOM
+
+  export CNI_COMMAND=ADD
+  assert invoke_sriov_cni
+  assert_file_contains "${DEFAULT_CNI_DIR}/enp175s0f1.calls" "LinkSetVfVlanQosProto enp175s0f1 0 1234 0 33024"
+
+  export CNI_COMMAND=DEL
+  assert invoke_sriov_cni
+  assert 'ip netns exec test_root_ns ip link show enp175s6'
+  assert_file_contains "${DEFAULT_CNI_DIR}/enp175s0f1.calls" "LinkSetVfVlanQosProto enp175s0f1 0 0 0 33024"
+}
+
+invoke_sriov_cni() {
+  echo "$CNI_INPUT" | ip netns exec test_root_ns go run "${this_folder}/sriov_mocked.go"
+}
+
+# Create a container and its related network namespace. The first parameter is
+# the name of the container, and as a convention, the netns name is `<container_name>_netns`
+make_container() {
```

Comment:
nit: update description as this no longer creates a container with podman

Reply:
done, I've renamed the method among delete_container()